pkg/unity/find-line-mappings.go

```diff
@@ -0,0 +1,50 @@
+package unity
+
+import (
+	"fmt"
+	"github.com/bugsnag/bugsnag-cli/pkg/utils"
+	"path/filepath"
+)
+
+// GetAndroidLineMapping locates the LineNumberMappings.json file for Android builds.
+//
+// This function attempts to resolve the path to the IL2CPP line number mapping file,
+// used for symbolication or debugging. The resolution follows this order:
+//
+//  1. If the input 'path' is non-empty, it is returned as-is.
+//  2. It checks the default path: Library/Bee/artifacts/Android/il2cppOutput/cpp/Symbols/LineNumberMappings.json.
+//  3. If not found, it searches under a backup folder (whose name ends with
+//     "BackUpThisFolder_ButDontShipItWithYourGame" inside 'projectRoot') for the same file.
+//
+// Parameters:
+//
+//	path        - an optional explicit path to the mapping file. If provided, it is returned directly.
+//	projectRoot - the root path of the Unity project, used to search for backup artifacts.
+//
+// Returns:
+//
+//	mappingPath - the resolved path to LineNumberMappings.json, or an empty string if not found.
+//	error       - non-nil if there was an error during backup folder resolution.
+func GetAndroidLineMapping(path string, buildDir string) (string, error) {
+	if path != "" {
+		return path, nil
+	}
+
+	// Check default artifacts path
+	defaultPath := filepath.Join(buildDir, "Library", "Bee", "artifacts", "Android", "il2cppOutput", "cpp", "Symbols", "LineNumberMappings.json")
```

Comment:
Do we understand what "Bee" is in this structure? I just wonder how likely it is to change. Perhaps we should recursive-search the "Library" and "BackUpThisFolder_ButDontShipItWithYourGame" subdirectories and return the first of that name?